src/wazuh_modules/syscollector/network/networkWindowsWrapper.h

```diff
@@ -0,0 +1,449 @@
+            if (interfaceAddress)
+            {
+                retVal = interfaceAddress->IpMask.String;
+            }
+        }
+        
+        return retVal;
+    }
+
+    std::string netmaskV6() const override
+    {
+        // "Convertion m_currentUnicastAddress->OnLinkPrefixLength to hex notation"
+        // This is not working currently. It always returns: "ffff:ffff:ffff:ffff::"
+        // Leaving this as empty for now.
+        // Windows XP netmask IPv6 is not supported
+        return "";
+    }
+
+    std::string broadcast() const override
+    {
+        std::string retVal { "unknown" };
+        const auto address { this->address() };
+        const auto netmask { this->netmask() };        
+        if (address.size() && netmask.size())
+        {
+            retVal = Utils::NetworkWindowsHelper::broadcastAddress(address, netmask);
+        }
+        return retVal;        
+    }
+
+    std::string broadcastV6() const override
+    {
+        return "unknown";
+    }
+
+    std::string gateway() const override
+    {
+        std::string retVal { "unknown" };
+        if (Utils::isVistaOrLater())
+        {
+            const auto gatewayAddress { m_interfaceAddress->FirstGatewayAddress };
+            if (gatewayAddress)
+            {
+                const auto gatewayFamily { gatewayAddress->Address.lpSockaddr->sa_family };
+                const auto sockAddress   { gatewayAddress->Address.lpSockaddr };
+                if (AF_INET == gatewayFamily)
+                {
+                    retVal = Utils::NetworkWindowsHelper::IAddressToString(gatewayFamily, 
+                                                                           (reinterpret_cast<sockaddr_in*>(sockAddress))->sin_addr);
+                }
+                else if (AF_INET6 == gatewayFamily)
+                {
+                    retVal = Utils::NetworkWindowsHelper::IAddressToString(gatewayFamily, 
+                                                                           (reinterpret_cast<sockaddr_in6*>(sockAddress))->sin6_addr);
+                }
+            }
+        }
+        else
+        {
+            // Under Windows XP, the only way to retrieve IPv4 gateway addresses is through GetAdaptersInfo()
+            PIP_ADDR_STRING currentGWAddress { nullptr };
+            PIP_ADAPTER_INFO currentAdapterInfo { m_adapterInfo };
+            bool foundMatch { false };
+            while (currentAdapterInfo && !foundMatch)
+            {
+                if (!(MIB_IF_TYPE_LOOPBACK == currentAdapterInfo->Type))
+                {
+                    if (currentAdapterInfo->Index == m_interfaceAddress->IfIndex)
+                    {
+                        // Found an interface match.
+                        currentGWAddress = &(currentAdapterInfo->GatewayList);
+                        while(currentGWAddress) 
+                        {
+                            retVal = currentGWAddress->IpAddress.String;
```

Comment:
this while will return only the last IpAddress string of the chained gateway list, is that the expected behavior?

Reply:
Good catch. I'll fix this.